pkg/cloud/azure/metadata.go

```diff
@@ -0,0 +1,58 @@
+package azure
+
+import (
+	"crypto/md5" //nolint:gosec
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/treeverse/lakefs/pkg/cloud"
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+type MetadataProvider struct {
+	logger logging.Logger
+}
+
+func NewMetadataProvider(logger logging.Logger) *MetadataProvider {
+	return &MetadataProvider{logger: logger}
+}
+
+type instanceMetadataResponse struct {
+	Compute struct {
+		SubscriptionId string `json:"subscriptionId"`
+	} `json:"compute"`
+}
+
+func (m *MetadataProvider) GetMetadata() map[string]string {
+	var PTransport = &http.Transport{Proxy: nil}
+	client := http.Client{Transport: PTransport}
```

Comment:
Why do we need to set no proxy? - the default uses the environment to capture this information which is recommended.